litmus-portal/frontend/src/components/Targets/ConnectTarget/index.tsx

```diff
@@ -1,28 +1,89 @@
 import { Typography } from '@material-ui/core';
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useMutation, useLazyQuery } from '@apollo/client';
+import { useSelector } from 'react-redux';
 import { history } from '../../../redux/configureStore';
-import ButtonFilled from '../../Button/ButtonFilled';
 import ButtonOutline from '../../Button/ButtonOutline';
 import TargetCopy from '../TargetCopy';
 import useStyles from './styles';
 import Scaffold from '../../../containers/layouts/Scaffold';
+import {
+  CreateClusterInput,
+  CreateClusterInputResponse,
+  Cluster,
+} from '../../../models/graphql/clusterData';
+import { USER_CLUSTER_REG, GET_CLUSTER } from '../../../graphql';
+import { RootState } from '../../../redux/reducers';
+import Loader from '../../Loader';
+import ButtonFilled from '../../Button/ButtonFilled';
 import Unimodal from '../../../containers/layouts/Unimodal';
 
-const engineUrl: string = 'kubectl apply -f';
-const CenteredTabs = () => {
+const ConnectTarget = () => {
   const classes = useStyles();
-  const [open, setOpen] = React.useState(false);
-  const handleClose = () => {
-    history.push('/workflows');
-    setOpen(false);
-  };
-  const handleClickModal = () => {
-    setOpen(true);
-  };
+  const [open, setOpen] = React.useState(true);
+  const [link, setLink] = React.useState('');
+  const [id, setID] = React.useState('');
+  const [modal, setModal] = React.useState(false);
+
   const handleClick = () => {
     history.push('/targets');
   };
+
+  const handleClose = () => {
+    history.push('/targets');
+  };
+
+  const selectedProjectID = useSelector(
+    (state: RootState) => state.userData.selectedProjectID
+  );
+
+  const [createClusterReg] = useMutation<
+    CreateClusterInputResponse,
+    CreateClusterInput
+  >(USER_CLUSTER_REG, {
+    onCompleted: (data) => {
+      const ID: string = data.userClusterReg.cluster_id;
+      const linkYaml: string = data.userClusterReg.token;
+      setLink(linkYaml);
+      setID(ID);
+    },
+  });
+
+  const [getCluster] = useLazyQuery(GET_CLUSTER, {
+    onCompleted: (data) => {
+      if (data && data.getCluster.length !== 0) {
+        data.getCluster.forEach((e: Cluster) => {
+          if (id === e.cluster_id) {
+            if (e.is_cluster_confirmed === true) {
+              setOpen(false);
+              setModal(true);
+            }
+          }
+        });
+      }
+    },
+    fetchPolicy: 'cache-and-network',
+    pollInterval: 5000,
```

Comment:
can you also add a poll interval to the check on schedule workflow stepper? cc: @Jonsy13 @rajdas98

Reply:
I think, it won't be required while scheduling, because we are only showing those clusters, which are active. cc: @oumkale